tests/aws/services/sns/test_sns.py

```diff
@@ -395,15 +395,21 @@ def test_unknown_topic_publish(self, sns_create_topic, snapshot, aws_client):
 
     @markers.aws.validated
     def test_topic_publish_another_region(
-        self, sns_create_topic, snapshot, aws_client, aws_client_factory, secondary_region_name
+        self,
+        sns_create_topic,
+        snapshot,
+        aws_client,
+        aws_client_factory,
+        region_name,
+        secondary_region_name,
     ):
         # create the topic in the default region, so that it's easier to clean up with the fixture
         topic_arn = sns_create_topic()["TopicArn"]
 
         # create a client in another region
         sns_client_region_2 = aws_client_factory.get_client(
             service_name="sns",
-            region_name=secondary_region_name,
+            region_name=secondary_region_name if region_name != "ap-southeast-1" else "us-east-2",
```

Comment:
I don't understand this change. Why `ap-southeast-1`?

Reply:
For now we just have randomised `TEST_AWS REGION_NAME` which is the primary region getting randomly generated and not the `SECONDARY_TEST_AWS_REGION_NAME`.

The default value we set for our tests of SECONDARY_TEST_AWS_REGION_NAME is `ap-southeast-1 `, so currently it is possible in the circleci workflow to have these regions same. Its already planned for the next iteration to add randomly generated secondary region in the workflow.

Reply:
Should we maybe update your change to check against `SECONDARY_TEST_AWS_REGION_NAME` then and not hardcoded `ap-southeast-1`? So it will also work if we change `SECONDARY_TEST_AWS_REGION_NAME` somehow?